tests/components/group/test_init.py

```diff
@@ -367,72 +320,150 @@ def test_group_updated_after_device_tracker_zone_change(self):
 
-    def test_reloading_groups(self):
-        """Test reloading the group config."""
-        assert setup_component(
-            self.hass,
-            "group",
-            {
-                "group": {
-                    "second_group": {"entities": "light.Bowl", "icon": "mdi:work"},
-                    "test_group": "hello.world,sensor.happy",
-                    "empty_group": {"name": "Empty Group", "entities": None},
-                }
-            },
-        )
 
-        group.Group.create_group(
-            self.hass, "all tests", ["test.one", "test.two"], user_defined=False
-        )
+async def test_is_on(hass):
+    """Test is_on method."""
+    hass.states.async_set("light.Bowl", STATE_ON)
+    hass.states.async_set("light.Ceiling", STATE_OFF)
 
-        assert sorted(self.hass.states.entity_ids()) == [
-            "group.all_tests",
-            "group.empty_group",
-            "group.second_group",
-            "group.test_group",
-        ]
-        assert self.hass.bus.listeners["state_changed"] == 1
-        assert len(self.hass.data[TRACK_STATE_CHANGE_CALLBACKS]["hello.world"]) == 1
-        assert len(self.hass.data[TRACK_STATE_CHANGE_CALLBACKS]["sensor.happy"]) == 1
-        assert len(self.hass.data[TRACK_STATE_CHANGE_CALLBACKS]["light.bowl"]) == 1
-        assert len(self.hass.data[TRACK_STATE_CHANGE_CALLBACKS]["test.one"]) == 1
-        assert len(self.hass.data[TRACK_STATE_CHANGE_CALLBACKS]["test.two"]) == 1
-
-        with patch(
-            "homeassistant.config.load_yaml_config_file",
-            return_value={
-                "group": {"hello": {"entities": "light.Bowl", "icon": "mdi:work"}}
-            },
-        ):
-            common.reload(self.hass)
-            self.hass.block_till_done()
-
-        assert sorted(self.hass.states.entity_ids()) == [
-            "group.all_tests",
-            "group.hello",
-        ]
-        assert self.hass.bus.listeners["state_changed"] == 1
-        assert len(self.hass.data[TRACK_STATE_CHANGE_CALLBACKS]["light.bowl"]) == 1
-        assert len(self.hass.data[TRACK_STATE_CHANGE_CALLBACKS]["test.one"]) == 1
-        assert len(self.hass.data[TRACK_STATE_CHANGE_CALLBACKS]["test.two"]) == 1
-
-    def test_modify_group(self):
-        """Test modifying a group."""
-        group_conf = OrderedDict()
-        group_conf["modify_group"] = {"name": "friendly_name", "icon": "mdi:work"}
-
-        assert setup_component(self.hass, "group", {"group": group_conf})
-
-        # The old way would create a new group modify_group1 because
-        # internally it didn't know anything about those created in the config
-        common.set_group(self.hass, "modify_group", icon="mdi:play")
-        self.hass.block_till_done()
+    assert group.is_on(hass, "group.none") is False
+    assert await async_setup_component(hass, "light", {})
+    assert await async_setup_component(hass, "group", {})
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6259301)